docs/java-rest/high-level/apis/search.asciidoc

```diff
@@ -178,3 +191,51 @@ cases need to be case accordingly:
 --------------------------------------------------
 include-tagged::{doc-tests}/SearchDocumentationIT.java[search-hits-singleHit-source]
 --------------------------------------------------
+
+===== Accessing Aggregations
+
+Aggregations can be retrieved from the `SearchResponse` by first getting the
+root of the aggregation tree, the `Aggregations` object, and then getting the
+aggregation by using its name. The following example shows how to retrieve the
+`by_company` terms aggregation, getting the buckets that is keyed with `Elastic`
+and then get the `average_age` sub-aggregation that is nested in this bucket: 
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-aggregations-get]
+--------------------------------------------------  
+
+Not that if you access aggregations by name like this, you need to specify the
+aggregation interface according to the type of aggregation you requested,
+otherwise a `ClassCastException` will be thrown:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-aggregations-get-wrongCast]
+--------------------------------------------------
+<1> This will throw an exception because "by_company" is a `terms` aggregation
+but we try to retrieve it as a `range` aggregation 
+
+It is also possible to access all aggregations as a map that is keyed by the
+aggregation name. In this case, the cast to the proper aggregation interface
+needs to happen explicitely:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-aggregations-asMap]
+--------------------------------------------------
+
+There are also getters that return all top level aggregations as a list:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-aggregations-asList]
+--------------------------------------------------  
+
+And last but not least you can iterate over all aggregations and then e.g.
+decide how to further process them based on their type:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-aggregations-iterator]
+--------------------------------------------------  
```

Comment:
All these different ways of reading aggs tend to confuse me a bit. Maybe we can split this in little parapgraphs?